plugins/Toolbox/src/Toolbox.py

```diff
@@ -0,0 +1,613 @@
+                            author["package_count"] = 0
+
+                        for package in self._metadata["packages"]:
+                            if package["author"]["name"] == author["name"]:
+                                author["package_count"] += 1
+                                author["type"] = package["package_type"]
+                                if "icon_url" in package:
+                                    author["icon_url"] = package["icon_url"]
+
+                    self._models["authors"].setMetadata(self._metadata["authors"])
+                    self.metadataChanged.emit()
+
+                    if not self._models["materials_showcase"]:
+                        self._models["materials_showcase"] = AuthorsModel(self)
+                    # TODO: Replace this with a proper API call:
+                    self._models["materials_showcase"].setMetadata(self._metadata["materials_showcase"])
+
+                    # This part is also needed for comparing downloaded packages to
+                    # installed packages.
+                    self._models["packages"].setMetadata(self._metadata["packages"])
+                    self._models["packages"].setFilter({"type": "plugin"})
+
+                    self.metadataChanged.emit()
+
+                    self.setViewPage("overview")
+                    return
+
+                except json.decoder.JSONDecodeError:
+                    Logger.log("w", "Toolbox: Received invalid JSON for package list.")
+                    return
+
+            if reply.url() == self._request_urls["plugins_showcase"]:
+                try:
+                    json_data = json.loads(bytes(reply.readAll()).decode("utf-8"))
+
+                    # Check for errors:
+                    if "errors" in json_data:
+                        for error in json_data["errors"]:
+                            Logger.log("e", "%s", error["title"])
+                        return
+
+                    # Create packages model with all packages:
+                    if not self._models["plugins_showcase"]:
+                        self._models["plugins_showcase"] = PackagesModel()
+                    self._metadata["plugins_showcase"] = json_data["data"]
+                    self._models["plugins_showcase"].setMetadata(self._metadata["plugins_showcase"])
+                    self.metadataChanged.emit()
+
+                    self.setViewPage("overview")
+                    return
+
+                except json.decoder.JSONDecodeError:
+                    Logger.log("w", "Toolbox: Received invalid JSON for showcase.")
+                    return
+
+
+        else:
+            # Ignore any operation that is not a get operation
+            pass
+
+    def _onDownloadProgress(self, bytes_sent: int, bytes_total: int):
+        if bytes_total > 0:
+            new_progress = bytes_sent / bytes_total * 100
+            self.setDownloadProgress(new_progress)
+            if new_progress == 100.0:
```

Comment:
Better have a habit of not comparing floats with an exact number using ==. In this case I'd rather use just bytes_sent and bytes_total.

Reply:
Fixed in 3ea80c7c7